pkg/solana/chain_reader_test.go

```diff
@@ -35,3 +66,596 @@ func TestSolanaChainReaderService_ServiceCtx(t *testing.T) {
+								Type:       config.ProcedureTypeAnchor,
+							},
+						},
+					},
+					MethodReturningUint64: {
+						AnchorIDL: fmt.Sprintf(baseIDL, uint64BaseTypeIDL, ""),
+						Procedures: []config.ChainReaderProcedure{
+							{
+								IDLAccount: "SimpleUint64Value",
+								Type:       config.ProcedureTypeAnchor,
+								OutputModifications: codeccommon.ModifiersConfig{
+									&codeccommon.PropertyExtractorConfig{FieldName: "I"},
+								},
+							},
+						},
+					},
+					DifferentMethodReturningUint64: {
+						AnchorIDL: fmt.Sprintf(baseIDL, uint64BaseTypeIDL, ""),
+						Procedures: []config.ChainReaderProcedure{
+							{
+								IDLAccount: "SimpleUint64Value",
+								Type:       config.ProcedureTypeAnchor,
+								OutputModifications: codeccommon.ModifiersConfig{
+									&codeccommon.PropertyExtractorConfig{FieldName: "I"},
+								},
+							},
+						},
+					},
+					MethodReturningUint64Slice: {
+						AnchorIDL: fmt.Sprintf(baseIDL, uint64SliceBaseTypeIDL, ""),
+						Procedures: []config.ChainReaderProcedure{
+							{
+								IDLAccount: "Uint64Slice",
+								Type:       config.ProcedureTypeAnchor,
+								OutputModifications: codeccommon.ModifiersConfig{
+									&codeccommon.PropertyExtractorConfig{FieldName: "Vals"},
+								},
+							},
+						},
+					},
+					MethodReturningSeenStruct: {
+						AnchorIDL: fmt.Sprintf(baseIDL, testStructIDL, strings.Join([]string{midLevelStructIDL, innerStructIDL}, ",")),
+						Procedures: []config.ChainReaderProcedure{
+							{
+								IDLAccount: "TestStruct",
+								Type:       config.ProcedureTypeAnchor,
+								OutputModifications: codeccommon.ModifiersConfig{
+									&codeccommon.HardCodeModifierConfig{OffChainValues: map[string]any{"ExtraField": AnyExtraValue}},
+									// &codeccommon.RenameModifierConfig{Fields: map[string]string{"NestedStruct.Inner.IntVal": "I"}},
+								},
+							},
+						},
+					},
+				},
+			},
+			AnySecondContractName: {
+				Methods: map[string]config.ChainDataReader{
+					MethodReturningUint64: {
+						AnchorIDL: fmt.Sprintf(baseIDL, uint64BaseTypeIDL, ""),
+						Procedures: []config.ChainReaderProcedure{
+							{
+								IDLAccount: "SimpleUint64Value",
+								Type:       config.ProcedureTypeAnchor,
+								OutputModifications: codeccommon.ModifiersConfig{
+									&codeccommon.PropertyExtractorConfig{FieldName: "I"},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (r *chainReaderInterfaceTester) GetChainReader(t *testing.T) types.ChainReader {
+	client := new(mockedRPCClient)
+	svc, err := solana.NewChainReaderService(logger.Test(t), client, r.conf)
+	if err != nil {
+		t.Logf("chain reader service was not able to start: %s", err.Error())
+		t.FailNow()
+	}
+
+	if r.reader == nil {
+		r.reader = &wrappedTestChainReader{
+			test:   t,
+			tester: r,
+		}
+	}
+
+	r.reader.service = svc
+	r.reader.client = client
+
+	return r.reader
+}
+
+type wrappedTestChainReader struct {
```

Comment:
Please add the TODO comment with a ticket number for replacing this with an actual Solana test validator :).

Reply:
Ahh, in the code. Will do